manifests/reverse_proxy.pp

```diff
@@ -1,9 +1,27 @@
 # Adds http reverse-proxy to parent conf
+#
+# @param path
+#   The path where to mount the reverse proxy
+# @param url
+#   The URL to forward to
+# @param servername
+#   The Apache vhost server name to listen on
+# @param port
+#   The port to listen on
+# @param ssl_protocol
+#   The ssl protocol(s) to accept
+# @param vhost_params
+#   Any parameters to pass to the apache::vhost resource
+# @param proxy_pass_params
+#   Any parameters to pass to the apache::vhost resource
 class foreman_proxy_content::reverse_proxy (
-  $path = '/',
-  $url  = "${foreman_proxy_content::foreman_url}/",
-  $port = $foreman_proxy_content::reverse_proxy_port,
-  $ssl_protocol = undef,
+  String $path = '/',
```

Comment:
Is there a better type to denote "an absolute (looking) path"?

Reply:
There is `Stdlib::Unixpath` which is strictly speaking not a URL path, but might be good enough.

Reply:
Yeah, I thought about Stdlib::Absolutepath, but yours is better.